src/lib.rs

```diff
@@ -766,6 +827,30 @@ impl<K: EnrKey> Enr<K> {
         Ok(())
     }
 
+    /// Removes a key from the ENR.
+    pub fn remove_key(&mut self, content_key: impl AsRef<[u8]>, enr_key: &K) -> Result<(), Error> {
+        self.content.remove(content_key.as_ref());
+
+        // add the new public key
+        let public_key = enr_key.public();
+        let mut pubkey = BytesMut::new();
+        public_key.encode().as_ref().encode(&mut pubkey);
+        self.content.insert(public_key.enr_key(), pubkey.freeze());
+
+        // increment the sequence number
+        self.seq = self
+            .seq
+            .checked_add(1)
+            .ok_or(Error::SequenceNumberTooHigh)?;
+
+        // sign the record
+        self.sign(enr_key)?;
```

Comment:
This will leave a corrupted Enr in the error case, so you need to either do the dance of having a back up or document this in the docs

Reply:
Yep, nice catch. 

This is also true for the surrounding other functions. I've had to modify most of the other functions to avoid the corrupt ENR path.